ordered_product/tests.py

```diff
@@ -3,58 +3,44 @@
     @pytest.mark.django_db()
-    def test_filter_by_user_name(self, ordered_product0, save_objects, client1, client0):
-        ordered_product0.user_name = client0
+    def test_filter_by_user_name(self, ordered_product0, saved_client0, saved_client1):
+        ordered_product0.user_name = saved_client0
         ordered_product0.save_ordered_product()
-        assert ordered_product0 not in list(OrderedProduct.filter_user_name(client1))
-        assert ordered_product0 in list(OrderedProduct.filter_user_name(client0))
+        assert ordered_product0 not in list(OrderedProduct.filter_user_name(saved_client1))
+        assert ordered_product0 in list(OrderedProduct.filter_user_name(saved_client0))
 
     @pytest.mark.django_db()
-    def test_filter_by_supplier_product_id(self, ordered_product0, supplier_product0, supplier_product1, save_objects):
-        ordered_product0.supplier_product_id = supplier_product0
+    def test_filter_by_supplier_product_id(self, ordered_product0, saved_supplier_product1):
+        ordered_product0.supplier_product_id = saved_supplier_product1
         ordered_product0.save_ordered_product()
-        assert ordered_product0 not in list(OrderedProduct.filter_supplier_product_id(supplier_product1))
-        assert ordered_product0 in list(OrderedProduct.filter_supplier_product_id(supplier_product0))
+        assert ordered_product0 in list(OrderedProduct.filter_supplier_product_id(saved_supplier_product1))
 
     @pytest.mark.django_db()
-    def test_filter_delivery_location_id(self, ordered_product0, delivery_location0, delivery_location1, save_objects):
-        ordered_product0.delivery_location_id = delivery_location0
+    def test_filter_delivery_location_id(self, ordered_product0, delivery_location1):
+        delivery_location1.add_delivery_location()
+        ordered_product0.delivery_location_id = delivery_location1
         ordered_product0.save_ordered_product()
-        assert ordered_product0 not in list(OrderedProduct.filter_delivery_location_id(delivery_location1))
-        assert ordered_product0 in list(OrderedProduct.filter_delivery_location_id(delivery_location0))
+        assert ordered_product0 in list(OrderedProduct.filter_delivery_location_id(delivery_location1))
 
     @pytest.mark.django_db()
-    def test_validators(self, delivery_location0, client0, supplier_product0, save_objects):
-        temp_ordered_product = OrderedProduct(
-                delivery_location_id=delivery_location0,
-                user_name=client0,
-                supplier_product_id=supplier_product0,
-                quantity=-2)
+    def test_validators(self, ordered_product0):
+        ordered_product0.quantity = -2
         with pytest.raises(ValidationError):
-            temp_ordered_product.save_ordered_product()
-        assert temp_ordered_product not in OrderedProduct.objects.all()
+            ordered_product0.save_ordered_product()
+        assert ordered_product0 not in OrderedProduct.objects.all()
```

Comment:
no need for line 40, the test checks validators, in line 39 validation error will be raised so the test is over.